user/views.py

```diff
@@ -1,30 +1,73 @@
+from django.shortcuts   import render
+from django.views       import View
+from django.http        import JsonResponse, HttpResponse
 
-from user.models            import User
-from my_settings            import SECRET_KEY, ALGORITHM
+from user.models        import User
+from my_settings        import SECRET_KEY,ALGORITHM
+from user.utils         import login_decorator
 
-class SignInView(View):
+
+PASSWORD_MIN_LENGTH     = 6
+USER_NAME_MIN_LENGTH    = 3
+USER_NAME_MAX_LENGTH    = 32
+
+class UserView(View):
     def post(self, request):
         data = json.loads(request.body)
         
         try:
             user_name   = data['user_name']
             password    = data['password']
+            nickname    = data['nickname']
+            email       = data['email']
+
+            if User.objects.filter(user_name=user_name).exists():
+                return JsonResponse({'message':'USER_NAME_ALREADY_EXISTS'}, status=409)
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({'message':'EMAIL_ALREADY_EXISTS'}, status=409)
+
+            if len(user_name) < USER_NAME_MIN_LENGTH:
+                return JsonResponse({'message':'SHORT_ID'}, status=409)
+            if len(user_name) > USER_NAME_MAX_LENGTH:
+                return JsonResponse({'message':'LONG_ID'}, status=409)
+            if len(password) < PASSWORD_MIN_LENGTH:
+                return JsonResponse({'message':'SHORT_PASSWORD'}, status=409)
             
-            if not User.objects.filter(user_name=user_name).exists():
-                return JsonResponse({'message':'INVALID_USER'}, status=401)
-            
-            user = User.objects.get(user_name=user_name)
+            user = User.objects.create(
+                    user_name   = user_name,
+                    password    = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode(),
+                    nickname    = nickname,
+                    email       = email
+                )
 
-            if bcrypt.checkpw(password.encode('utf-8'), user.password.encode('utf-8')):
-                access_token = jwt.encode({'user_id':user.id}, SECRET_KEY, algorithm=ALGORITHM)
+            access_token   = jwt.encode({'user_id':user.id}, SECRET_KEY, algorithm=ALGORITHM)
 
-                return JsonResponse({'message':'SUCCESS', "access_token":access_token}, status=200)
-            return JsonResponse({'message':'SIGNIN_FAIL'}, status=401)
+            return JsonResponse({'message':'SUCCESS','access_token':access_token}, status=200)
         
         except KeyError:
             return JsonResponse({'message':'INVALID_KEYS'}, status=400)
+    
+    @login_decorator
+    def patch(self, request):
+        print(request)
+        try:
+            if not request.body:
+                return JsonResponse({'message':'SUCCESS'}, status=200)
+
+            data = json.loads(request.body)
+
+            user    = User.objects.get(id=request.user.id)
```

Comment:
데코레이터 활용해서 user 객체 받으셨군요! good good!

```suggestion
            user    = request.user
```
request에 들어있는 `user`이라는 속성이 이미, django User 객체입니다 :)
왜냐하면 `login decorator`에서
```python
            user            = User.objects.get(id=payload['user_id'])
            request.user    = user
```
이렇게 request.user에 할당해준 객체가 User객체이기 때문입니다 :) 이해가 혹시 되셨을까요?
따라서 views파일에서는 request에 들어있는 user 인스턴스를 바로 사용하실 수 있습니다

Reply:
네 이해완료했습니다!